src/Adapter/Product/Stock/Update/ProductStockUpdater.php

```diff
@@ -118,6 +128,57 @@ public function update(ProductId $productId, ProductStockProperties $properties,
         }
     }
 
+    /**
+     * Resets product stock to zero, both Product and associated StockAvailable are reset, and a stock movement linked to
+     * the employee from context is generated.
+     *
+     * @param ProductId $productId
+     * @param ShopConstraint $shopConstraint
+     *
+     * @throws CoreException
+     * @throws ProductStockException
+     */
+    public function resetStock(ProductId $productId, ShopConstraint $shopConstraint): void
+    {
+        if ($shopConstraint->forAllShops()) {
+            $shops = $this->productRepository->getAssociatedShopIds($productId);
+        } else {
+            $shops = [$shopConstraint->getShopId()];
+        }
```

Comment:
What about groups? 🤔

Reply:
I'm not sure either... 

For now this method is always given a "all stop" context, and the product page cannot be accessed in group shop, maybe it's the reason ?

@jolelievre if you could enlighten us ? 🤔

Reply:
Product entity cannot be edited in the shop group context which is why only single and All shops context are handled
In other parts of the code I added a check though that throws an exception when a group constraint is used, I'll add this safety here as well!

Reply:
Even if technically for now the method is only called with an All shops constraint, I built in a way that can be potentially used for other use cases I didn't want to explicitly bind it to the All shops context

So it's a lot better to add this extra check indeed!